### ISSUE TITLE
Mac App Store build rejected for Private API usage [2.0.10]

### ISSUE THREAD

**Issue by UzEE** [electron/electron#17263]
* **Electron Version:** 
  * 2.0.10 (however, building the app with 4.0.7 and dumping symbols (via `nm -u Electron\ Framework`) used in Electron Framework shows the private symbol is still included in 4.0.7)

#### App A
We have recently found that your app uses or references the following non-public APIs:

Found private symbol usage.
Symbol: _close$NOCANCEL
From framework: /usr/lib/libSystem.B.dylib
In binary: Contents/Frameworks/Electron Framework.framework/Versions/A/Electron Framework

We are constantly reevaluating and identifying non-public APIs that you may have been using for an extended period of time. You should always use public APIs and frameworks and ensure they are up-to-date to prevent this issue in the future.

The use of non-public APIs is not permitted on the App Store, because it can lead to a poor user experience should these APIs change.

#### App B

We have recently found that your app uses or references the following non-public APIs:

Found private symbol usage.
Symbol: _CGDisplayUsesForceToGray
From framework: /System/Library/Frameworks/CoreGraphics.framework/Versions/A/CoreGraphics
Symbol: __CFIsObjC
From framework: /System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation
In binary: Contents/Frameworks/Electron Framework.framework/Versions/A/Electron Framework

We are constantly reevaluating and identifying non-public APIs that you may have been using for an extended period of time. You should always use public APIs and frameworks and ensure they are up-to-date to prevent this issue in the future.

The use of non-public APIs is not permitted on the App Store, because it can lead to a poor user experience should these APIs change.

### Additional Information
We've been using Electron v2.0.10 in our apps for several months now and have never had any of our client apps rejected because of private API usage before, but now two of our apps have been rejected for private API usage within a week. Both apps were submitted from different publisher accounts, and apart from cosmetic changes are essentially identical, however the rejection cites different private APIs in both cases.

Additionally, going through the issues here, it seems like the private APIs referenced in App B should already have been fixed for Electron v2.0.10 based on #13938.

To confirm if these symbols were indeed being referenced by Electron Framework, I dumped the symbols referenced in Electron Framework file using the following:

```
nm -u Electron\ Framework > symbols.txt
```

I only found the symbol `_close$NOCANCEL` in both apps and the symbols because of which App B was rejected weren't in the framework file for either of the apps (which makes sense since their usage is supposed to be removed after #13938), so it's odd that Apple somehow detected their usage in the app and rejected it.

I've also tried building the app with Electron v4.0.7 and check it's symbol usage, and `_close$NOCANCEL` is still present in the Electron Framework of the latest version. I haven't tried submitting the v4.0.7 to the store since 3.x branch introduced some breaking changes in our apps and we'll have to resolve them before we can upgrade electron.

**Comment by miniak** [comment]
Related to https://cs.chromium.org/chromium/src/base/mac/close_nocancel.cc?q=close_implementation&sq=package:chromium&g=0&l=56

**Comment by miniak** [comment]
I've checked:
- `electron-v3.1.9-mas-x64`
- `electron-v4.2.0-mas-x64`
- `electron-v5.0.1-mas-x64`
- `electron-v6.0.0-beta.2-mas-x64`

All of these are linking to `_close$NOCANCEL`. None of the uses `CGDisplayUsesForceToGray` or `_CFIsObjC`.

**Comment by miniak** [comment]
@UzEE please make sure that you're using the latest version for a given release branch.

**Comment by sofianguy** [comment]
Thank you for taking the time to report this issue and helping to make Electron better.

The version of Electron you reported this on has been superseded by newer releases. 

If you're still experiencing this issue in Electron v4.2.x or later, please add a comment specifying the version you're testing with and any other new information that a maintainer trying to reproduce the issue should know.

I'm setting the `blocked/need-info` label for the above reasons. This issue will be closed 7 days from now if there is no response.

Thanks in advance! Your help is appreciated.
